plan/planbuilder.go

```diff
@@ -219,6 +219,10 @@ func (b *planBuilder) buildSet(v *ast.SetStmt) Plan {
 			IsSystem: vars.IsSystem,
 		}
 		if _, ok := vars.Value.(*ast.DefaultExpr); !ok {
+			if cn, ok2 := vars.Value.(*ast.ColumnNameExpr); ok2 && cn.Name.Table.L == "" {
```

Comment:
Why should we do this?

Reply:
To handle the case like `set sql_mode = strict_all_tables` which is consistent with MySQL,
but query like `set @a = a` should have caused an error.

This is not addition action, but move-equivalent action, which will not affect the feature of TiDB.